### PR TITLE
Track originalViewValue

### PR DESCRIPTION
**Pull request by nmccready**
This is needed to no longer blindly set the view back to `''` on canceling.

IE:

`ctrl.$setViewValue('');` is not `ctrl.$setViewValue(originalViewValue);`

### CODE REVIEW

File: src/validator.directive.js [hunk 1]
```diff
@@ -268,6 +268,10 @@
          */
         var uid = ctrl.validationId = guid();
 
+        /**
+         * to have avalue to rollback to
```
Comment (huei90) [inline src/validator.directive.js]: opps typo

File: Gruntfile.js [hunk 2]
```diff
@@ -112,4 +112,5 @@ module.exports = function(grunt) {
   grunt.registerTask('build', ['check', 'clean', 'concat', 'uglify', 'copy']);
   grunt.registerTask('test', ['build', 'karma']);
   grunt.registerTask('cov', ['coveralls']);
+  grunt.registerTask('default', ['build', 'test']);
```
Comment (huei90) [inline Gruntfile.js]: Do you mind remove this line because it out of scope for this PR.

**Comment by huei90** [comment]
Great and awesome @nmccready 

Everything looks good 😄 There is only one extra needed

- [ ] Test it!

File: Gruntfile.js (hunk 2 above)
Comment (nmccready) [inline Gruntfile.js]: Sure I'll remove this.

**Comment by nmccready** [comment]
Fixed everything. 

> Great and awesome @nmccready
> 
> Everything looks good 😄 There is only one extra needed
> 
>  Test it!

Not sure if this mean you require specs or not. Seems overkill.

**Comment by huei90** [comment]
Ok! Merge it now. thanks again 🙇

**Pull request merged**